### PR TITLE
QEMU: Add 'dirty-bitmap' and 'pbs-state', handle page_size more consistently

### PR DESCRIPTION
**Pull request by cstation**
In rare cases, QEMU suspend-to-disk files contain sections named `dirty-bitmap` and `pbs-state`. These sections are defined at the beginning of the file, after the `ram`-section is defined. After the definitions of the sections (`QEVM_SECTION_START`s), the `QEVM_SECTION_PART`s belonging to the `ram`-section follow. This means that the sections do not occur once in total, but they occur in multiple parts accross the file. Therefore, the information about the sections needs to be stored and accessible throughout the parsing process.

Moreover, the actual offsets of the section parts are calculated in two ways, and the `page_size` variable is not used consistently. This PR fixes these issues.

### CODE REVIEW

File: volatility3/framework/layers/qemu.py
Comment (ikelos) [inline volatility3/framework/layers/qemu.py]: Is there ever a time this value could be 0?  I'm trying to figure out why (even in our original code), we set a default of None and then or it with 4096.  I'd probably just set the default of `4096`, but there's the very specific situation where `page_size` is 0 that this catches but a default of `4096` would not...

File: volatility3/framework/layers/qemu.py
Comment (ikelos) [inline volatility3/framework/layers/qemu.py:87]: This makes an implicit assumption that the `page_size` will always be a power of 2.  I don't really have a problem with that, it's almost certainly always going to be true, but it might be nice just to include a comment that we're making that assumption.  Not a showstopper, just a nice to have...  5:)

**Review by ikelos — COMMENTED** [review]
Awesome, thanks very much for catching those gotchas!  5:)  It all looks good, there's a couple of points I'm pondering but nothing that blocks the merge.  Let me know if you'd like to deal with them before I merge this, otherwise I'll merge it in a couple of days.  5:)

File: volatility3/framework/layers/qemu.py
Comment (cstation) [reply volatility3/framework/layers/qemu.py]: I can't imagine a situation where `page_size` would be 0, since data is stored in chunks of `page_size` bytes. The only thing that could happen is that the value would be set to 0 due to errors while reading the configuration JSON, but that should (I guess) be handled in the function `_read_configuration`. Would you like me to change the `None` to `4096`?

File: volatility3/framework/layers/qemu.py
Comment (cstation) [reply volatility3/framework/layers/qemu.py:87]: Didn't realize that :) I will add a comment!

File: volatility3/framework/layers/qemu.py
Comment (ikelos) [reply volatility3/framework/layers/qemu.py]: Errr, yeah, I think it would be easier to read, but it occurs to me we could just force `_read_configuration` to populate the `page_size` entry whether there was one or not.  We could also deal with errors there too?  If you'd be happy doing that, then I think it might be the best way of dealing with it (if you'd prefer me to make the changes, just shout).  Thanks!  5:)

File: volatility3/framework/layers/qemu.py
Comment (ikelos) [reply volatility3/framework/layers/qemu.py:87]: Perfect, thank you!  5:D

**Review by ikelos — APPROVED** [review]
Looks good, thanks very much!  5:D

File: volatility3/framework/layers/qemu.py
Comment (cstation) [reply volatility3/framework/layers/qemu.py]: Issue is fixed now I think (after my force-pushed commit). An exception is raised if reading the configuration fails, and if no JSON is found at the end of the file (= no config is present), an emtpy dict is returned. I slightly prefer leaving the config as close to the original, rather than forcefully populating it. :relieved:

**Pull request merged**
